.github/workflows/macos.yml

```diff
@@ -42,28 +41,40 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      - name:  Prepare brew and compiler caches
+        id:    prep-caches
+        shell: bash
+        run: |
+          set -eu
+          BREW_DIR="$(brew --cache)"
+          DISCARD_DIR="${{ github.workspace }}/discard"
+          mkdir -p "$DISCARD_DIR"
+          mv -f "$BREW_DIR"/* "$DISCARD_DIR"
+          mkdir -p "$CCACHE_DIR"
+          echo "::set-output name=brew_dir::$BREW_DIR"
+          echo "::set-output name=ccache_dir::$CCACHE_DIR"
+          echo "::set-output name=today::$(date +%F)"
+          echo "::set-output name=name_hash::$(echo '${{ matrix.conf.name }}' | shasum | cut -b-8)"
+
```

Comment:
I decided to put this "prep caches" task together in one for brew and ccache.

We could have these being separate tasks, Prepare brew cache vs. Prepare ccache cache, and it would add a handful more lines to the YAML.

The benefit is we can share the "today date" value, instead of repeating it in two places. I also thing the code is simple and relatable in it's purpose that it can live together.

Reply:
Grouping these together makes sense in my opinion - whenever I read/edit workflow files, I tend to glaze over the caching part anyway.